### PR TITLE
fix(ansible): Adding bios config job and reboot tasks in bios.yml

### PR DESCRIPTION
**Pull request by rahulsachdev**

### CODE REVIEW

File: ansible/bios.yml [hunk 1]
```diff
@@ -26,11 +26,18 @@
...
       register: result
+      tags: ["TEST"]
 
     - name: Debug print bios serial number
       ansible.builtin.debug: msg={{ result.redfish_facts.bios_attribute.entries[0][1].SerialNumber | default(result.redfish_facts.bios_attribute.entries[0][1].SystemServiceTag) }}
 
-    # TODO: configre BIOS to be always on ( see lab/hardware/dh123) and any virtualization or hyper threading settings we might need
+
+    - name: Debug print bios serial number
+      when: inventory_hostname == 'dh1bmc'
```
Comment (glimchb) [inline ansible/bios.yml:36]: remove `when` here and rename the task from `serial`

File: ansible/bios.yml [hunk 2]
```diff
@@ -43,25 +50,28 @@
...
-#    - name: Create BIOS configuration job (schedule BIOS setting update)
-#      community.general.idrac_redfish_command:
-#       category: Systems
-#       command: CreateBiosConfigJob
-#       baseuri: "{{ ansible_host }}"
-#       username: "{{ ansible_user }}"
-#       password: "{{ ansible_password }}"
-#      when: bios_attribute.changed
+    - name: Create BIOS configuration job (schedule BIOS setting update)
+      when: inventory_hostname == 'dh1bmc'
```
Comment (glimchb) [inline ansible/bios.yml:58]: I think we should use another condition here
In previous task or from gather_facts we can already know if we talking to iDRAC or not
let's use that as a condition

File: ansible/bios.yml [hunk 3]
```diff
@@ -43,25 +50,28 @@
...
-#    - name: Reboot system to apply new BIOS settings
-#      community.general.redfish_command:
-#       category: Systems
-#       command: PowerReboot
-#       baseuri: "{{ ansible_host }}"
-#       username: "{{ ansible_user }}"
-#       password: "{{ ansible_password }}"
-#      when: bios_attribute.changed
+    - name: Reboot system to apply new BIOS settings
+      when: inventory_hostname == 'dh1bmc'
```
Comment (glimchb) [inline ansible/bios.yml:69]: two `when` statements ? one here and another one below `bios_config_job.changed`

**Review by glimchb — CHANGES_REQUESTED** [review]
2 when statements

File: ansible/bios.yml [hunk 4]
```diff
@@ -26,11 +26,17 @@
...
+        #tags: ["TEST"]
 
     - name: Debug print bios serial number
       ansible.builtin.debug: msg={{ result.redfish_facts.bios_attribute.entries[0][1].SerialNumber | default(result.redfish_facts.bios_attribute.entries[0][1].SystemServiceTag) }}
 
-    # TODO: configre BIOS to be always on ( see lab/hardware/dh123) and any virtualization or hyper threading settings we might need
+        # TODO: configre BIOS to be always on ( see lab/hardware/dh123) and any virtualization or hyper threading settings we might need
+        #- name: Debug print bios attributes
+        #ansible.builtin.debug: msg={{ result }}
+        #ansible.builtin.debug: msg={{ result.redfish_facts.bios_attribute.entries[0][1].SystemManufacturer }}
```
Comment (glimchb) [inline ansible/bios.yml:37]: this is a good debug line to keep uncommented
all the rest comments should be removed

File: ansible/bios.yml [hunk 5]
```diff
@@ -43,25 +49,30 @@
        username: "{{ ansible_user }}"
        password: "{{ ansible_password }}"
       register: bios_attribute
-      tags: ["TEST"]
+        #      tags: ["TEST"]
```
Comment (glimchb) [inline ansible/bios.yml:52]: remove this line

File: ansible/bios.yml [hunk 6]
```diff
@@ -43,25 +49,30 @@
...
-#    - name: Create BIOS configuration job (schedule BIOS setting update)
-#      community.general.idrac_redfish_command:
-#       category: Systems
-#       command: CreateBiosConfigJob
-#       baseuri: "{{ ansible_host }}"
-#       username: "{{ ansible_user }}"
-#       password: "{{ ansible_password }}"
-#      when: bios_attribute.changed
+    - name: Create BIOS configuration job (schedule BIOS setting update)
+      when: result.redfish_facts.bios_attribute.entries[0][1].SystemManufacturer is defined and result.redfish_facts.bios_attribute.entries[0][1].SystemManufacturer == "Dell Inc." and bios_attribute.changed
```
Comment (glimchb) [inline ansible/bios.yml:58]: let's split into lines like
https://github.com/opiproject/lab/blob/dabdb551b0f0f0188206114af003ad8d9577bb3a/ansible/monitoring.yml#L72-L76

**Review by glimchb — COMMENTED** [review]
Much better!!!
few small suggestions inside
and please squash all the commits to a single one

**Comment by rahulsachdev** [comment]
Will work on Boris's suggestions and submit a new PR

**Pull request closed without merging**